File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260915-110924-39025/analysis/acceptance-project/src/main/java/customerimporter/CUIT.java
package customerimporter;

import java.util.Arrays;
import java.util.List;
import java.util.function.Supplier;

public class CUIT extends Identification {

    public static final String TYPE = "C";

    private static final List<String> validHeaders = Arrays.asList("20", "23", "24", "25", "26", "27", "30", "33", "34");

    // error messages

    public static String invalidCUITNumberErrorDescription() {
        return "Invalid CUIT number";
    }

    // initialization

    CUIT(String aNumber) {
        super(aNumber);
    }

    // identification

    public String type() {
        return TYPE;
    }

    // type

    public boolean isCUIT() {
        return true;
    }

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return cuitNumber();
    }

    public String cuitNumber() {
        return number();
    }

    // validation

    protected void assertIsValid() {
        String cuitNumber = number();

        // "23-25666777-9" size 13
        assertThat(cuitNumber.length() >= 12 && cuitNumber.length() <= 13);
        assertThat(cuitNumber.charAt(2) == '-' && cuitNumber.charAt(cuitNumber.length() - 2) == '-');
        assertThat(validHeaders.contains(cuitNumber.substring(0, 2)));
        assertThat(Character.isDigit(cuitNumber.charAt(cuitNumber.length() - 1)));
        assertThat(Characters.areAllDigits(cuitNumber.substring(3, cuitNumber.length() - 2)));
    }

    protected String invalidNumberErrorDescription() {
        return invalidCUITNumberErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-203929-73794/output/project/src/customerimporter/cuit.py
from customerimporter.customer_identification import CustomerIdentification


class Cuit(CustomerIdentification):

    # type

    @classmethod
    def type_code(cls):
        return "C"

    # identification

    def cuit_number_if_none(self, a_none_block):
        return self.number()

    def is_cuit(self):
        return True

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid CUIT number"

    # validation

    @classmethod
    def valid_headers(cls):
        return ["20", "23", "24", "25", "26", "27", "30", "33", "34"]

    @classmethod
    def assert_is_valid(cls, an_identification_number):
        # "23-25666777-9" size 13
        cls._assert_that(len(an_identification_number) >= 12 and len(an_identification_number) <= 13)
        cls._assert_that(an_identification_number[2] == "-" and an_identification_number[-2] == "-")
        cls._assert_that(an_identification_number[:2] in cls.valid_headers())
        cls._assert_that(an_identification_number[-1].isdigit())
        cls._assert_that(cls._are_all_digits(an_identification_number[3:-2]))
